src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -1414,14 +1414,7 @@ bool ICollection.IsSynchronized
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         object ICollection.SyncRoot
         {
-            get
-            {
-                // As a struct, this instance will be boxed in order to call this property getter.
-                // We will return the boxed instance. If the caller boxes and reboxes repeatedly,
-                // they'll get a different SyncRoot each time.
-                // No real alternative, but we don't anticipate this will be much problem anyway.
-                return this;
-            }
+            get { throw new NotSupportedException(); }
```

Comment:
It can't be used directly, and I don't think the text of concurrent collection's exception makes sense for this immutable one ("The SyncRoot property may not be used for the synchronization of concurrent collections.").  I also don't think rewritten text would add much value over the default text associated with NotSupportedException ("Specified method is not supported"), which is why I didn't add another resource to customize the message.  But my care factor is fairly low on this one. :)